### DISCUSSION TITLE
Row that restricts field queries cannot be lexiconized

### DISCUSSION THREAD

**Discussion by breezechime** [sqlalchemy/sqlalchemy#11723] [Usage Questions]
### Describe the bug

When I want to query a certain model, once I restrict the fields, the returned value will not be the model but the Row. I cannot understand this, which makes many methods of my model class unable to be called, and all methods of this Row are private, indicating that it is an immutable property. This leads to significant issues with the advantages of ORM, which seems to be different from other ORM frameworks. The most shocking thing is that I have been unable to find a public method to convert Row to dict. Because its methods are all private, the official does not want to call it, so how can we convert it into a dict

当我想查询某个模型时，一旦限制字段，返回的将不是模型而是Row，我无法理解，这让我的模型类很多方法无法被调用，并且这个Row所有方法都是私有的将说明它是一个不可修改的属性，这导致ORM的优势有大大问题，其他ORM框架似乎不是这样的，最让人震惊是我一直找不到的方法公开让Row转成dict。因为它的方法都是私有的说明官方不希望调用它，所以如何才能将它转成dict呢

### SQLAlchemy Version in Use

2.0.31

### DBAPI (i.e. the database driver)

pymysql

### Database Vendor and Major Version

sqlite3

### Python Version

3.11

### Operating system

OSX

### To Reproduce

```python
class User(Base):
    """用户"""
    __tablename__ = 'user'

    id = Column(Integer, primary_key=True, autoincrement=True)
    username = Column(String(32), nullable=False, unique=True)
    nickname = Column(String(32))


Base.metadata.create_all(engine)

user = User(id=1, username='test1', nickname='李四2')

row = user.query(User.nickname).filter_by(id=1).first()
print(dict(row))
```

### Error

```
print(dict(row))
          ^^^^^^^^^
ValueError: dictionary update sequence element #0 has length 3; 2 is required
```

**Comment by CaselIT** [comment]
hi,

You are selecting a single column, not the user class, so the returned values are tuple like objects of a single element, that's the column you selected. 
If you use select you can get dict-like object using https://docs.sqlalchemy.org/en/20/core/connections.html#sqlalchemy.engine.Result.mappings, but since this is query that does not return result objects, you can use https://docs.sqlalchemy.org/en/20/core/connections.html#sqlalchemy.engine.Row._asdict
